drivers/adc/ad7616/ad7616.c

```diff
@@ -506,6 +844,99 @@ int32_t ad7616_core_setup(struct ad7616_dev *dev)
 
 	return 0;
 }
+#endif
+
+/**
+ * @brief Setup the sequencer layers
+ * @param dev The device structure
+ * @param layers Arrays of layers
+ * @param layers_nb Nubers of items in the layers array.
+ * @param burst 1 if needed, 0 otherwise
+ * @return 0 on success, an error value otherwise
+ */
+int32_t ad7616_setup_sequencer(struct ad7616_dev *dev,
+			       struct ad7616_sequencer_layer *layers,
+			       uint32_t layers_nb, uint8_t burst)
+{
+	int ret;
+	for (uint32_t i = 0; i < layers_nb; i++) {
+		uint16_t data = 0;
+
+		// Last one ?
+		if (i == layers_nb - 1)
+			data |= AD7616_SSREN;
+		data |= (layers[i].ch_b - AD7616_VB0) << 4;
+		data |= layers[i].ch_a;
+
+		ret = ad7616_write(dev, AD7616_REG_SEQUENCER_STACK(i), data);
+		if (ret != 0)
+			return ret;
+	}
+
+	if (burst) {
+		ret = ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_BURSTEN,
+					AD7616_BURSTEN);
+		if (ret != 0)
+			return ret;
+	}
+
+	// Enable channel sequencer
+	ret = ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_SEQEN,
+				AD7616_SEQEN);
+	if (ret != 0)
+		return ret;
+
+	dev->layers_nb = layers_nb;
+
+	// Dummy CONV pulse as asked in datasheet
+	return ad7616_toggle_conv(dev);
+}
+
+int32_t ad7616_disable_sequencer(struct ad7616_dev *dev)
+{
+	dev->layers_nb = 1;
+	return ad7616_write_mask(dev, AD7616_REG_CONFIG,
+				 AD7616_SEQEN | AD7616_BURSTEN, 0);
```

Comment:
it is kind of strange that for enabling you are using AD7616_SEQEN and for disabling it is simply 0. it is also kind of strange that you are using AD7616_SEQEN for mask and enable bit as well.
I would suggest to make things a little bit more clear for the reviewer by creating two different macros:
AD7616_SEQEN_MASK for the mask which can remain (1 << 5)
and AD7616_SEQEN(x) which can be ((x & 1)<< 5)
and then use it something like this:
```
ad7616_write_mask(dev, AD7616_REG_CONFIG, AD7616_SEQEN_MASK, AD7616_SEQEN(1));
```
similarly for AD7616_BURSTEN